core/drivers/cbmem_console.c

```diff
@@ -6,20 +6,35 @@
 #include <compiler.h>
 #include <console.h>
 #include <drivers/cbmem_console.h>
+#include <drivers/serial.h>
 #include <io.h>
 #include <keep.h>
 #include <kernel/dt.h>
 #include <libfdt.h>
 #include <mm/core_mmu.h>
+#include <types_ext.h>
 #include <util.h>
 
 #define CURSOR_MASK (BIT(28) - 1)
 #define OVERFLOW BIT(31)
 
+struct cbmem_console {
+	uint32_t size;
+	uint32_t cursor;
+	uint8_t body[0];
+} __packed;
+
+struct cbmem_console_data {
+	paddr_t base;
+	struct cbmem_console *console;
+	struct serial_chip chip;
+	uint32_t size;
+};
+
 /*
  * Structures describing coreboot's in-memory descriptor tables. See
- * https://github.com/coreboot/coreboot/blob/master/src/commonlib/include/commonlib/coreboot_tables.h
- * for canonical implementation.
+ * coreboot's src/commonlib/include/commonlib/coreboot_tables.h for
+ * details.
```

Comment:
last fixup reverts a change requested by @jforissier!

Reply:
So what would satisfy the permanent link you're looking for? Do you mean something like this? (this is a GitHub permalink)

https://github.com/coreboot/coreboot/blob/ea2a38be323173075db3b13729a4006ea1fef72d/src/commonlib/include/commonlib/coreboot_tables.h

Reply:
looks ok to me.
sorry for bothering with such details.

Reply:
No problem, thanks for the feedback. :)